### PR TITLE
CurrentInputMethodLanguageTag

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): #721 

## PR Type
What kind of change does this PR introduce?

- Feature 

## What is the current behavior?
Missing implementation of `CurrentInputMethodLanguageTag`

## What is the new behavior?
Implemented `CurrentInputMethodLanguageTag` and `TrySetInputMethodLanguageTag`. 

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tested code with current [supported SDKs](../README.md#supported)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/nventive/Uno/blob/master/doc/.feature-template.md). (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](doc/articles/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [x] Contains **NO** breaking changes
- [x] Updated the [Release Notes](https://github.com/nventive/Uno/tree/master/doc/ReleaseNotes)
- [x] Associated with an issue (GitHub or internal)

## Other information
**OS support** - I have noticed that Android tends not to report the Input Keyboard Language reliably and it has been confirmed based on what I read. The newer `LanguageTag` property I am using seems to be even less reliable than the deprecated `Locale` property, unfortunately.

**Code duplication** - I had to resort to code duplication in `Language.iOS.cs` - `UIView FindFirstResponder(UIView view)` method is the same code as in `Uno.UI.Extensions.UIViewExtensions`, which I could not use here as it is in the `Uno.UI` library and hence not accessible. I could move `UIViewExtensions` to `Uno.UWP`, but that would mean breaking backward compatibility. Please advise on the best course of action.

Internal Issue (If applicable):

### CODE REVIEW

**Comment by MartinZikmund** [comment]
@carldebilly Is the code duplication mentioned in the `Other Information` section OK or should it be addressed somehow?

**Comment by carldebilly** [comment]
I'll let @jeromelaban express himself on this one :-).

But I'm curious: what is the need/use case for this feature?

**Comment by MartinZikmund** [comment]
@carldebilly this feature is needed for Telerik controls support (#721), so I thought it would be nice to implement it for Android and iOS as well, although arguably getting current keyboard input language is usually not too useful 😊

**Comment by yowl** [comment]
@carldebilly seems like there is something special about french keyboards and the minus sign.  This code is used in Telerik's numeric text box control to allow negative numbers
```
        private static bool IsAzertyKeyboard
        {
            get
            {
                return Windows.Globalization.Language.CurrentInputMethodLanguageTag.StartsWith("fr-");
            }
        }
```
and
```
            if (RadNumericBox.IsAzertyKeyboard)
            {
                isNegativeSign = isNegativeSign || (key == VirtualKey.Number6 && DeviceTypeHelper.GetDeviceType() != DeviceType.Phone);
            }
```

**Comment by jeromelaban** [comment]
Indeed, there's duplication there, but that's not a big issue. We'll eventually consolidate, but for now, you can put that method in a `ViewExtensions.iOSmacOS.cs` class in the Extensions folder, so we know that it is reused at that level. We'll call one from the other to avoid binary breaking changes.

I've made changes to that class for macOS support here, to make it available on both: https://github.com/nventive/Uno/pull/802/files#diff-cce17a30f244931e5ac882cfed84883d

File: src/Uno.UWP/Globalization/Language.Android.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+		{
+			get
+			{
+				var inputMethodManager = ContextHelper.Current.GetSystemService(Context.InputMethodService).JavaCast<InputMethodManager>();
+				var currentInputMethod = inputMethodManager.CurrentInputMethodSubtype;
+				return currentInputMethod.LanguageTag;
+			}
+		}
+
+		public static bool TrySetInputMethodLanguageTag(string languageTag) => false;
```
Comment (jeromelaban) [inline src/Uno.UWP/Globalization/Language.Android.cs]: We may want to log that it is not supported.

File: src/Uno.UWP/Globalization/Language.wasm.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+using System.Collections.Generic;
+using System.Text;
+
+namespace Windows.Globalization
+{
+	public partial class Language
+	{
+		public static string CurrentInputMethodLanguageTag { get; private set; } = "";
+
+        public static bool TrySetInputMethodLanguageTag(string languageTag)
```
Comment (jeromelaban) [inline src/Uno.UWP/Globalization/Language.wasm.cs]: We'll probably need to adjust that to get the browser cultures, but for now, not throwing is best :)

File: src/Uno.UWP/Globalization/Language.Android.cs (hunk 1 above)
Comment (MartinZikmund) [reply src/Uno.UWP/Globalization/Language.Android.cs]: The `false` should actually be ok - according to docs - https://docs.microsoft.com/en-us/uwp/api/windows.globalization.language.trysetinputmethodlanguagetag - it is expected that the method does nothing on all devices except IoT.

Comment (jeromelaban) [reply src/Uno.UWP/Globalization/Language.Android.cs]: That makes sense, thanks.

File: src/Uno.UWP/Globalization/Language.Android.cs [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+namespace Windows.Globalization
+{
+	public partial class Language
+	{
+		public static string CurrentInputMethodLanguageTag
+		{
+			get
+			{
+				var inputMethodManager = ContextHelper.Current.GetSystemService(Context.InputMethodService).JavaCast<InputMethodManager>();
```
[the remark this answers is not in the document]
Comment (MartinZikmund) [reply src/Uno.UWP/Globalization/Language.Android.cs]: Fixed :-)

**Comment by MartinZikmund** [comment]
Resolved the `UIViewExtensions` code duplication, although now there is a potential name clash between `Uno.Extensions.UIViewExtensions` and `UIKit.UIViewExtensions`.

**Comment by MartinZikmund** [comment]
I merged latest master, are any additional changes needed here? @carldebilly @jeromelaban

**Comment by ghuntley** [comment]
Thank-you @MartinZikmund. There was some CI issues which have since been resolved.

**Pull request merged**